pycbc/inference/models/gaussian_noise.py

```diff
@@ -852,7 +853,46 @@ def _nowaveform_loglr(self):
             setattr(self._current_stats, '{}_optimal_snrsq'.format(det), 0.)
         return -numpy.inf
 
-    def _loglr(self):
+    @property
+    def multi_signal_support(self):
+        """ The list of classes that this model supports in a multi-signal
+        likelihood
+        """
+        return [type(self)]
+
+    def multi_loglikelihood(self, models):
+        """ Calculate a multi-model (signal) likelihood
+        """
+        # This could be made somewhat faster by presumming the waveforms
+        # to save on all the extra inner products, this implementations
+        # is somehwat overexplicit. Some checks could also be done once.
+        #It should be OK, for the common case
+        # where waveform generation is the dominant cost.
+
+        loglr = 0
+        # handle sum[<d|h_i> - 0.5 <h_i|h_i>]
+        for m in models:
+            loglr += m._loglr(save_waveforms=True)
+
+        # finally add in the lognl term from this model
+        for m1, m2 in itertools.combinations(models, 2):
+            if not hasattr(m1, 'wfs') or not hasattr(m2, 'wfs'):
+                raise FailedWaveformError
+
+            for det in m1.wfs:
+                h1, slc1 = m1.wfs[det]
+                h2, slc2 = m2.wfs[det]
+
+                if slc1 != slc2:
+                    raise RuntimeError('gaussian noise implementation of multi'
+                                       'signal model requires all frequency'
+                                       'cutoffs to be the same!')
+
+                h1h2 = h1[slc1].inner(h2[slc1]).real  # < h1, h2>
+                loglr += - h1h2  # This is -0.5 * (<h1|h2> + <h2|h1>)
+        return loglr + self.lognl
+
```

Comment:
@ahnitz Wouldn't it be easier to just sum up the waveforms, then pass that through the normal loglr or loglikelihood function?

Reply:
Ok, done.